src/components/DescriptionView.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { mount } from '@cypress/react';
+import { Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import { ThemeProvider } from '@mui/styles';
+import createMockStore from 'redux-mock-store';
+import React from 'react';
+import DescriptionView from './DescriptionView';
+import { DatasetRoles } from '../constants';
+import globalTheme from '../modules/theme';
+import history from '../modules/hist';
+
+const userRoles = [DatasetRoles.STEWARD];
+const initialState = {
+  dataset: {
+    description: 'A dataset description',
+  },
+};
+
+beforeEach(() => {
+  const mockStore = createMockStore([]);
+  const store = mockStore(initialState);
+  mount(
+    <Router history={history}>
+      <Provider store={store} />
+      <ThemeProvider theme={globalTheme}>
+        <DescriptionView
+          description={initialState.dataset.description}
+          userRoles={userRoles}
+          updateDescriptionFn={(text: any) => {
+            initialState.dataset.description = text;
+          }}
+        />
+      </ThemeProvider>
+    </Router>,
+  );
+});
+
+describe('DescriptionView', () => {
```

Comment:
I have little-to-no practical experience in this repo / UI technologies, I appreciate how easy to follow your tests are!